test/mocks.cpp

```diff
@@ -59,6 +66,10 @@ pnal_eth_handle_t * mock_pnal_eth_init (
 
    handle = (pnal_eth_handle_t *)calloc (1, sizeof (pnal_eth_handle_t));
 
+   handle->if_name = if_name;
```

Comment:
Instead of allocating a pnal_eth_handle_t object in this function, perhaps one could initialise mock_os_data.eth_if_handle in mock_init() to some constant pointer, e.g (void*)42, and then return it here?

Reply:
Agreed. BTW, doesn't the calloc call leak memory for every test-case using mock_clear()?

Reply:
Yes, I assume so. I will update according to Fredriks idea.